pydantic_settings/main.py

```diff
@@ -33,7 +40,6 @@ class SettingsConfigDict(ConfigDict, total=False):
     env_parse_enums: bool | None
     cli_prog_name: str | None
     cli_parse_args: bool | list[str] | tuple[str, ...] | None
-    cli_settings_source: CliSettingsSource[Any] | None
```

Comment:
This was a bug. It's not possible to provide `cli_settings_source` as a configuration setting as the `settings_cls` cannot be defined (circular dependency).